File: 2015-07-09_ICML/collatz.jl
function collatz(x::Integer)
    i = 0
    while x != 1
        i += 1
        if rem(x, 2) == 0
            x = fld(x, 2)
        else
            x = 3 * x + 1
        end
    end
    return i
end

[collatz(x) for x in 1:10_000]

code_native(collatz, (Int, ))

File: 2015-07-09_ICML/collatz.py
def collatz(x):
    i = 0
    while x != 1:
        i += 1
        if x % 2 == 0:
            x = x / 2
        else:
            x = 3 * x + 1
    return i

[collatz(x) for x in range(1, 100001)]
